src/core/client/stream/tabs/Comments/ExternalImageInput/ExternalImageInput.tsx

```diff
@@ -55,39 +53,72 @@ const ExternalImageInput: FunctionComponent<Props> = ({ onSelect }) => {
   }, []);
 
   return (
-    <div className={styles.root}>
-      <HorizontalGutter>
-        <HorizontalGutter>
-          <Localized id="comments-postComment-pasteImage">
-            <InputLabel htmlFor="coral-comments-postComment-pasteImage">
-              Paste image URL
-            </InputLabel>
-          </Localized>
-          <Flex>
-            <TextField
-              id="coral-comments-postComment-pasteImage"
-              className={styles.input}
-              value={url}
-              onChange={onChange}
-              onKeyPress={onKeyPress}
-              fullWidth
-              variant="seamlessAdornment"
-              color="streamBlue"
-              ref={ref}
-            />
-            <Localized id="comments-postComment-insertImage">
-              <Button
-                color="stream"
-                onClick={onClick}
-                className={styles.insertButton}
-              >
-                Insert
-              </Button>
-            </Localized>
-          </Flex>
-        </HorizontalGutter>
-      </HorizontalGutter>
-    </div>
+    <Form
+      onSubmit={() => {
+        onSelect(url);
+      }}
+    >
+      {({ handleSubmit }) => (
+        <div className={styles.root}>
+            <Field
+              name="externalImg"
+              validate={getImageValidators()}
+            >
+              {({ input, meta }) => (
+                  <HorizontalGutter>
+                    <HorizontalGutter>
+                      <Localized id="comments-postComment-pasteImage">
+                        <InputLabel htmlFor="coral-comments-postComment-pasteImage">
+                          Paste image URL
+                        </InputLabel>
+                      </Localized>
+                      <Flex>
+                        <TextField
+                          name="externalImg"
+                          id="coral-comments-postComment-pasteImage"
+                          className={styles.input}
+                          onChange={(evt) => {
+                            setURL(evt.currentTarget.value);
+                            input.onChange(evt);
+                          }}
+                          onKeyPress={onKeyPress}
+                          fullWidth
+                          variant="seamlessAdornment"
+                          color="streamBlue"
+                          ref={ref}
+                        />
+                        <Localized id="comments-postComment-insertImage">
+                          <Button
+                            color="stream"
+                            disabled={!!meta.error}
+                            onClick={(e) => {
+                              if (!url || meta.error) return;
+
+                              handleSubmit();
+                            }}
```

Comment:
This suggestion will integrate FinalForm.

```suggestion
                            disabled={submitting || invalid || pristine}
                            onClick={handleSubmit}
```